spot-client/src/common/remote-control/P2PSignalingClient.js

```diff
@@ -0,0 +1,129 @@
+import { logger } from 'common/logger';
+import P2PSignalingBase from './P2PSignalingBase';
+
+
+/**
+ * A P2P signaling client which exposes method for sending remote control commands and is to be used by a Spot Remote.
+ */
+export default class P2PSignalingClient extends P2PSignalingBase {
+    /**
+     * Creates new instance.
+     *
+     * @param {P2PSignalingCallbacks} callbacks - See type description for more info.
+     * @param {P2PSignalingOptions} options - See type description for more info.
+     */
+    constructor(callbacks, options) {
+        super(callbacks, options);
+
+        /**
+         * @typedef {Object} PromiseLike
+         * @property {Function} resolve - hooked up to Promise's resolve method.
+         * @property {Function} reject - hooked up to Promise's reject method.
+         */
+        /**
+         * Map stores command requests sent over P2P signaling in order to resolve original promises when ack is
+         * received.
+         * @type {Map<number, PromiseLike>}
+         * @private
+         */
+        this._p2pSignalingRequests = new Map();
+    }
+
+    /**
+     * Client processing for data channel messages. A client processes only acks.
+     *
+     * @param {string} remoteAddress - The remote address that sent the message.
+     * @param {Object} msg - FIXME.
+     * @private
+     * @returns {void}
+     */
+    _processDataChannelMessage(remoteAddress, msg) {
+        const { command, requestId } = msg;
+
+        if (command === 'ack' && requestId) {
+            const commandPromise = this._p2pSignalingRequests.get(requestId);
+
+            if (commandPromise) {
+                commandPromise.resolve();
+            } else {
+                logger.warn('No P2P command promise exists', { requestId });
+            }
+
+            this._p2pSignalingRequests.delete(requestId);
+        } else if (msg.command === 'status' && msg.newStatus) {
+            this._callbacks.onStatusUpdateReceived(remoteAddress, msg.newStatus);
+        } else {
+            super._processDataChannelMessage(remoteAddress, msg);
```

Comment:
It doesn't, but it logs a warning which would be duplicated in both server and the client. Also is a place where you would put common msg processing in future. Because it's in future I can get rid of it if you insist.